File: solutions/go/clock/2/clock.go
// Package clock implements a simple library that implements
// methods to handle times without dates
package clock

import (
	"fmt"
)

// Clock struct to keep time in minutes
type Clock struct {
	timeInMinutes int
}

// New returns a new instance of Clock
func New(hours, minutes int) Clock {
	maxMinutes := 24 * 60
	totalMinutes := hours*60 + minutes
	return Clock{
		timeInMinutes: (maxMinutes + totalMinutes%maxMinutes) % maxMinutes,
	}
}

// Add minutes to a given Clock and returns a new instance
func (c Clock) Add(minutesToAdd int) Clock {
	return New(c.hours(), c.minutes()+minutesToAdd)
}

// Subtract minutes from a given Clock and returns a new instance
func (c Clock) Subtract(minutesToSubtract int) Clock {
	return c.Add(-minutesToSubtract)
}

// String returns time in HH:mm format
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hours(), c.minutes())
}

func (c Clock) hours() int {
	return c.timeInMinutes / 60
}

func (c Clock) minutes() int {
	return c.timeInMinutes % 60
}

File: solutions/csharp/clock/2/Clock.cs
public class Clock
{
    private readonly int timeInMinutes;

    public Clock(int hours, int minutes)
    {
        timeInMinutes = AdjustTimeInMinutes(hours * 60 + minutes);
    }

    public int Hours => timeInMinutes / 60;

    public int Minutes => timeInMinutes % 60;

    public Clock Add(int minutesToAdd) => new Clock(Hours, Minutes + minutesToAdd);

    public Clock Subtract(int minutesToSubtract) => Add(-minutesToSubtract);

    public override string ToString()
    {
        return $"{Hours:D2}:{Minutes:D2}";
    }

    public override bool Equals(object obj)
    {
        var other = obj as Clock;
        if (other == null)
        {
            return false;
        }

        return timeInMinutes == other.timeInMinutes;
    }

    private int AdjustTimeInMinutes(int minutesIn)
    {
        var maxMinutes = 24 * 60;
        return (maxMinutes + minutesIn % maxMinutes) % maxMinutes;
    }
}